### ISSUE TITLE
GraphQL: Transaction cursor doesn't support rollbacks

### ISSUE THREAD

**Issue by SebastienGllmt** [blockscout/blockscout#5321]
A common use-case for the API is to get all transactions for an address. This is possible with the graphql API thanks to #1100

However, one critical part of such a feature is pagination. Since blockchains may experience rollbacks, any pagination system has to be able to work even if a rollback occurs in-between pagination requests.

GraphQL natively has a standard for this called the [GraphQL Cursor Connections Specification](https://relay.dev/graphql/connections.htm) and blockscout uses a library called Absinthe.Relay that allows writing pagination that conforms to this specification

You can see an example of how pagination looks like currently below

```graphql
{
  address(hash: "address_here") {
    transactions(first:2) {
      edges {
        node {
          blockNumber
        }
        cursor
      }
    }
  }
}
```

```json
{
  "data": {
    "address": {
      "transactions": {
        "edges": [
          {
            "cursor": "YXJyYXljb25uZWN0aW9uOjA=",
            "node": {
              "blockNumber": 2766881
            }
          },
          {
            "cursor": "YXJyYXljb25uZWN0aW9uOjE=",
            "node": {
              "blockNumber": 2766804
            }
          }
        ]
      }
    }
  }
}
```

### Expected behaviour

If a rollback happens during pagination, the client should somehow be made aware of it. The simplest way is to just invalidate the cursor so that when they try and query the next page, they get an error saying the cursor is invalid. You can do this, for example, by encoding the <block_hash, tx_hash> inside the cursor so that if the transaction no longer exists inside the block, the cursor will no longer exist.

### Actual behaviour

If you [look at the graphql query](https://github.com/blockscout/blockscout/blob/b05c5c6ed2398cf21decc15a715b17f20d5aa428/apps/block_scout_web/lib/block_scout_web/resolvers/transaction.ex#L15), no custom cursor is defined. According to [Absinthe.Relay documentation](https://hexdocs.pm/absinthe_relay/Absinthe.Relay.Connection.html#module-supplying-edge-information),

>"If you provide a :cursor argument, then your value will override the internally generated cursor. This may or may not be desirable.

So this means that this graphql query is simply using the default cursor. If you look at [the PR Absinthe.Relay](https://github.com/absinthe-graphql/absinthe_relay/pull/173) source code, that added the ability to override the default cursor, they say that

> Currently the cursors in this package just encode the OFFSET and OFFSET + LIMIT.

If you look at their [current source code](https://github.com/absinthe-graphql/absinthe_relay/blob/d254162a2b7f5b23e6afa186da0f1879f6c484d0/lib/absinthe/relay/connection.ex#L572), it seems this behavior is unchanged (this is still their default cursor implementation)

tl;dr: the current cursor doesn't get invalidated by rollbacks and instead just silently return the wrong result.

### Extra note

If this issue is solved properly, it should allow making a GraphQL query that asks for all transactions that appear after some <block_hash, tx_hash> pair. This is what is needed for external feeds to be able to efficiently and reliably sync data using the blockscout API
